src/Deserializers/SnakDeserializer.php

```diff
@@ -137,8 +148,18 @@ private function deserializeDataValue( $serialization ) {
 	 */
 	private function deserializePropertyId( $serialization ) {
 		try {
-			return new PropertyId( $serialization );
-		} catch ( InvalidArgumentException $ex ) {
+			$id = $this->propertyIdParser->parse( $serialization );
+
+			if ( $id->getEntityType() !== Property::ENTITY_TYPE ) {
```

Comment:
This is a very very "hot" function… can we save a function call here by switching this out for "instanceof PropertyId"?

Reply:
I don't know if instanceof is actually faster. Want to benchmark?

Reply:
```
$ php maintenance/eval.php --wiki wikidatawiki
> $propertyId = new Wikibase\DataModel\Entity\PropertyId( 'P123' );

> $propertyId->getEntityType();

>  $t0 = microtime(1); for ( $i = 0; $i < 100000; $i++ ) { if ( $propertyId->getEntityType() !== Wikibase\DataModel\Entity\Property::ENTITY_TYPE ) { echo 'fail'; } }; echo microtime(1) - $t0;
0.061592102050781
>  $t0 = microtime(1); for ( $i = 0; $i < 100000; $i++ ) { if ( !$propertyId instanceof Wikibase\DataModel\Entity\PropertyId ) { echo 'fail'; } }; echo microtime(1) - $t0;
0.017740964889526
>  $t0 = microtime(1); for ( $i = 0; $i < 100000; $i++ ) { if ( $propertyId->getEntityType() !== Wikibase\DataModel\Entity\Property::ENTITY_TYPE ) { echo 'fail'; } }; echo microtime(1) - $t0;
0.056143999099731
>  $t0 = microtime(1); for ( $i = 0; $i < 100000; $i++ ) { if ( !$propertyId instanceof Wikibase\DataModel\Entity\PropertyId ) { echo 'fail'; } }; echo microtime(1) - $t0;
0.018042087554932
```

So yes, I think `instanceof` is actually quite a bit faster :)